src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -3130,10 +3130,12 @@ ScheduleBackgroundTask(int64 jobId, Oid owner, char *command, int dependingTaskC
 
 	/* 2. insert new task */
 	{
-		Datum values[Natts_pg_dist_background_task] = { 0 };
-		bool nulls[Natts_pg_dist_background_task] = { 0 };
+		TupleDesc tupleDescriptor = RelationGetDescr(pgDistBackgroundTask);
 
-		memset(nulls, true, sizeof(nulls));
+		Datum *values = (Datum *) palloc(tupleDescriptor->natts * sizeof(Datum));
+		bool *nulls = (bool *) palloc(tupleDescriptor->natts * sizeof(bool));
+
+		memset(nulls, true, tupleDescriptor->natts * sizeof(bool));
```

Comment:
let's simply zero-out them

```suggestion
		Datum *values = (Datum *) palloc0(tupleDescriptor->natts * sizeof(Datum));
		bool *nulls = (bool *) palloc0(tupleDescriptor->natts * sizeof(bool));
```